### PR TITLE
Implemented forecast extended information

### PR DESCRIPTION
**Pull request by FranciscoGaraventa**
When a card is pressed then it shows a FragmentDialog with all the information of forecast to that day.
Activity shows five cards corresponding to first our (00:00) of every forecast day.

Package tree:
image
image

Gif on card pressed action:
forecastDay

### CODE REVIEW

File: app/src/main/java/com.example.weatherapp/data/dialog/WeatherDayListener.kt [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package com.example.weatherapp.data.dialog
+
+interface WeatherDayListener {
```
Comment (SartiMau) [inline app/src/main/java/com.example.weatherapp/data/dialog/WeatherDayListener.kt]: as this listener is used in the adapter and for the adapter you can add it in the same file with it

File: app/src/main/java/com.example.weatherapp/mvp/presenter/WeatherPresenter.kt [hunk 2]
```diff
@@ -9,13 +10,21 @@ class WeatherPresenter(
     private val view: WeatherContract.View
 ) : WeatherContract.Presenter {
 
-    override fun getForecast() {
+    override fun getForecast(onForecastListener: WeatherDayListener) {
```
Comment (SartiMau) [inline app/src/main/java/com.example.weatherapp/mvp/presenter/WeatherPresenter.kt]: it not necessary that you pass this listener as a parameter, you can access it from the view directly

File: app/src/main/res/values/styles.xml [hunk 3]
```diff
@@ -6,4 +6,12 @@
         <item name="android:paddingTop">@dimen/padding_top_text_view_xsmall</item>
     </style>
 
+    <style name="CardTextViewStyleWithPaddingBottom" parent="CardTextViewStyle">
```
Comment (SartiMau) [inline app/src/main/res/values/styles.xml]: can't you unify these two styles?

**Review by SartiMau — CHANGES_REQUESTED** [review]

**Review by MarcosCGlobant — APPROVED** [review]
After Maus comments are fixed, LGTM

File: app/src/main/java/com.example.weatherapp/data/dialog/WeatherDayListener.kt (hunk 1 above)
Comment (FranciscoGaraventa) [reply app/src/main/java/com.example.weatherapp/data/dialog/WeatherDayListener.kt]: Removed a listener interface and added a function callback to WeatherAdapter.

File: app/src/main/java/com.example.weatherapp/mvp/presenter/WeatherPresenter.kt (hunk 2 above)
Comment (FranciscoGaraventa) [reply app/src/main/java/com.example.weatherapp/mvp/presenter/WeatherPresenter.kt]: Done

File: app/src/main/res/values/styles.xml (hunk 3 above)
Comment (FranciscoGaraventa) [reply app/src/main/res/values/styles.xml]: Unified text view styles

**Review by SartiMau — APPROVED** [review]

**Review by MarcosCGlobant — APPROVED** [review]

**Pull request merged**